README.md

```diff
@@ -16,10 +16,18 @@ Note: detailed go documentation is available at [godoc.org](https://godoc.org/gi
 ### Concepts
 `go-diskfs` has a few basic concepts:
 
+* Backend
 * Disk
 * Partition
 * Filesystem
 
+#### Backend
+Backend is a (relatively) thin layer which abstracts low-level read/write operations. Through a backend you can seamlessly operate different disk formats (see a bit stale qcow2 branch).
```

Comment:
Please get rid of the "bit stale qcow2 branch" comment. It is fine in a comment on a PR or issue, but not in long-lived docs.

Reply:
Also, we might want to make it clear that you can use a Backend or just use a disk directly?

Reply:
Well, we could add some notice like this in the backend section of readme:

> For convenience, if provided fs.File is not `backend`-like, it is internally wrapped into `raw` backend. But keep in mind, this will work with device files and raw images only! Opt to creating backend instance explicitely for the clarity sake.

Reply:
> Not exactly - it stands true only in the case of using raw images or device files...
> If provided fs.File is not a backend-like, it is internally wrapped into raw backend, but that provided fs.File might be not a raw image or device and in that case the user will eventually end up with an error.
> So, I'd propose using backends explicitly for clarity.

I don't understand. There still are the `Open(path string)` and `Create(path string)` functions, so you still can use a file directly? It is a convenience function equivalent to (pseudo-code) `Open(backed.FromFile(path)`

Reply:
> Well, we could add some notice like this in the backend section of readme

I didn't quite understand this either. What does "backend-like" mean, or internally wrapped?

Reply:
> What does "backend-like" mean

A type, that fully implement [backend.Storage](https://github.com/aol-nnov/go-diskfs/blob/open-file/backend/interface.go#L27) interface

>  or internally wrapped?

Here it is: https://github.com/aol-nnov/go-diskfs/blob/open-file/diskfs.go#L117-L121

Seems, I've chosen not the best terms for explanation, sorry..

Reply:
>> What does "backend-like" mean
> A type, that fully implements [backend.Storage](https://github.com/aol-nnov/go-diskfs/blob/open-file/backend/interface.go#L27) interface

Haha. OK, now I get it. In that case, you mean, "if the argument implements the `backend.Storage` interface, then..."

There are only several possibilities:

* call `OpenBackend(f fs.File)`
* call `Open(p string)` - this only accepts a path to a device or disk image
* call `Create(p string)` - this only accepts a path to a device or disk image

The latter two cases are just convenience wrappers around `OpenBackend(backend.FromFile(p))` (pseudo-code), so you are not referring to that, you must be referring to `OpenBackend(f fs.File)`. Why would we pass something that needs to be "wrapped"? I guess maybe if you have a read-only backend, e.g. I want to use an `fs.File` handle that I have that refers to some storage that has on it a disk image? I guess you are saying that `Disk` must have a `Backend` property, so you need to translate the passed `fs.File` into `Backend`?

Would that even work? Don't I need an actual `backend.File` at minimum?

Reply:
Well, to be frank, the more I read this, the more I dislike it... Moreover, func `OpenBackend(f fs.File..` looks somewhat contradictory to me - function name says "backend" and argument says "File", what?!

I think, it'll be more clear if we pass a backend as argument like `func OpenBackend(b backend.Storage, opts ...OpenOpt)`.

Reply:
Changed it that way https://github.com/aol-nnov/go-diskfs/blob/open-file/diskfs.go#L241

What do you think, looks better, eh? )

Reply:
It does.